utils/vim/syntax/swift.vim

```diff
@@ -101,8 +101,13 @@ syn region swiftType contained contains=swiftType,swiftParamDelim
       \ start="[^@](" end=")"
 syn match swiftParamDelim contained
       \ /,/
-
-syn region swiftTypeParameters start="<" end=">" contained
+" <Generic Clause> (generics)
+syn region swiftTypeParameters contained contains=swiftArchetype,swiftConstraint
+      \ start="<" end=">"
+syn match swiftArchetype contained skipwhite nextgroup=swiftColonType
+      \ /\<[A-Za-z_][A-Za-z_0-9]*\>/
+syn keyword swiftConstraint contained
+      \ where
```

Comment:
Note that constraints are being moved to the end of the declaration: https://github.com/apple/swift-evolution/blob/master/proposals/0081-move-where-expression.md